BL-5418 Sub-shelf formatting

gmartin7 opened the pull request:
* Make sourceName display smaller and based on
   existence of sourceName
* Move "Clear" link up so it stays on the same line
* Set bookshelf sourceName if "::" separator exists

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomlibrary/295)
<!-- Reviewable:end -->

andrew-polk commented:
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/modules/browse/browse.less, line 137 at r1](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/295#-L7zYxtr4mEuwm2Tojbb:-L7zYxtr4mEuwm2Tojbc:b-nq0ey7) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/7cad70096d007d0f12599865fe87951b01e3b788/src/app/modules/browse/browse.less#L137)):*
> ```Less
>   div.childShelf {
>     margin-left: 1em;
>     p {
> ```

Is there a reason to use p rather than div? div seems like it would be more consistent with the way we are doing things currently.
Either way, we should give it a class name.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/modules/browse/browse.less, line 137 at r1](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295#-L7zYxtr4mEuwm2Tojbb:-L7z_dJ4C05Ou9-PnDSf:b-r7bdbq) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/7cad70096d007d0f12599865fe87951b01e3b788/src/app/modules/browse/browse.less#L137)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Is there a reason to use p rather than div? div seems like it would be more consistent with the way we are doing things currently.
Either way, we should give it a class name.
</blockquote></details>

I was just following the suggestion in the issue:
[We will need to take an "English String" of something like "Nigeria::American University Of Nigeria" and produce something like

<a>Nigeria<p>American University of Nigeria</p></a>]
I am thinking of replacing 'sourceName' with 'orgName'. That seems clearer.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/modules/browse/browse.less, line 137 at r1](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295#-L7zYxtr4mEuwm2Tojbb:-L7zb6oF8ya_wVorY4N6:br1ogxp) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/7cad70096d007d0f12599865fe87951b01e3b788/src/app/modules/browse/browse.less#L137)):*
<details><summary><i>Previously, gmartin7 (Gordon Martin) wrote…</i></summary><blockquote>

I was just following the suggestion in the issue:
[We will need to take an "English String" of something like "Nigeria::American University Of Nigeria" and produce something like

<a>Nigeria<p>American University of Nigeria</p></a>]
I am thinking of replacing 'sourceName' with 'orgName'. That seems clearer.
</blockquote></details>

Hm. Actually, now that I take a second look, this is invalid html. You can't put block elements (div, p, etc.) inside inline elements (a, span, etc.).

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/modules/browse/browse.less, line 137 at r1](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295#-L7zYxtr4mEuwm2Tojbb:-L7zbpv11OGZTE_Znqxj:b-ked3re) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/7cad70096d007d0f12599865fe87951b01e3b788/src/app/modules/browse/browse.less#L137)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Hm. Actually, now that I take a second look, this is invalid html. You can't put block elements (div, p, etc.) inside inline elements (a, span, etc.).
</blockquote></details>

Interesting. Doing some more reading, this is considered valid in HTML5 (but not 4).
Still seems wrong and inconsistent, personally.
Regardless, we don't want to necessarily style every p (or div or span) inside a div.childShelf the same way, so we definitely should add a class name to specify when we want the smaller font.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/app/modules/browse/browse.less, line 137 at r1](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295#-L7zYxtr4mEuwm2Tojbb:-L7zbYVbDCvVmViMEhMQ:b-y5hhjd) ([raw file](https://github.com/bloombooks/bloomlibrary/blob/7cad70096d007d0f12599865fe87951b01e3b788/src/app/modules/browse/browse.less#L137)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

Interesting. Doing some more reading, this is considered valid in HTML5 (but not 4).
Still seems wrong and inconsistent, personally.
Regardless, we don't want to necessarily style every p (or div or span) inside a div.childShelf the same way, so we definitely should add a class name to specify when we want the smaller font.
</blockquote></details>

Yeah, I found the same thing you did re: HTML5. I changed srcName to orgName everywhere and added another safety check.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/bloombooks/bloomlibrary/295)*
<!-- Sent from Reviewable.io -->
